Python/Product/PythonTools/PythonTools/InterpreterList/InterpreterListToolWindow.cs

```diff
@@ -394,6 +410,60 @@ sealed class InterpreterListToolWindow : ToolWindowPane {
                 .Where(Directory.Exists));
         }
 
+        private void DeleteEnvironment_CanExecute(object sender, CanExecuteRoutedEventArgs e) {
+            var view = e.Parameter as EnvironmentView;
+            e.CanExecute = view?.CanBeDeleted == true;
+            e.Handled = true;
+        }
+
+        private void DeleteEnvironment_Executed(object sender, ExecutedRoutedEventArgs e) {
+            // TODO: this is assuming that all environments that CanBeDeleted are conda environments, which may not be true in the future
+            var view = e.Parameter as EnvironmentView;
+            var result = MessageBox.Show(
+                Resources.EnvironmentPathsExtensionDeleteConfirmation.FormatUI(view.Configuration.PrefixPath),
+                Resources.ProductTitle,
+                MessageBoxButton.YesNo,
+                MessageBoxImage.Question
+            );
+            if (result != MessageBoxResult.Yes) {
+                return;
+            }
+
+            var compModel = _site.GetService(typeof(SComponentModel)) as IComponentModel;
+            var registry = compModel.GetService<IInterpreterRegistryService>();
+            // TODO: reuse a single conda environment manager between this and
+            // the one in CondaExtension, since it can be slow to create
```

Comment:
Better off making it fast to create, or just make it a MEF export and then get it through the component model.

Reply:
Yeah, MEF with some caching of previously seen conda environments along with their `conda -V` version will be helpful. I'll file a new issue for it.